modules/dex/src/lib.rs

```diff
@@ -304,25 +306,25 @@ decl_module! {
 					!share_increment.is_zero() && !other_currency_increment.is_zero() && !base_currency_increment.is_zero(),
 					Error::<T>::InvalidLiquidityIncrement,
 				);
+				let new_total_shares = total_shares.checked_add(&share_increment).ok_or(Error::<T>::SharesOverflow)?;
 
 				T::Currency::transfer(other_currency_id, &who, &Self::account_id(), other_currency_increment)?;
 				T::Currency::transfer(base_currency_id, &who, &Self::account_id(), base_currency_increment)?;
-
 				Self::deposit_calculate_interest(other_currency_id, &who, share_increment);
-				<TotalShares<T>>::mutate(other_currency_id, |share| *share = share.saturating_add(share_increment));
-				<Shares<T>>::mutate(other_currency_id, &who, |share| *share = share.saturating_add(share_increment));
+				<TotalShares<T>>::insert(other_currency_id, new_total_shares);
+				<Shares<T>>::mutate(other_currency_id, &who, |share| *share += share_increment);
```

Comment:
add a comment to explain why this can't overflow (because total shares did not overflow and shares is always less or equal to total shares)